test/acceptance/user.controller.acceptance.ts

```diff
@@ -141,11 +141,17 @@ describe('UserController', () => {
   describe('authentication functions', () => {
     let plainPassword: string;
     let jwt_auth_service: JWTAuthenticationService;
+    let bcrypt_hasher: BcryptHasher;
 
     before('create new user', async () => {
+      bcrypt_hasher = new BcryptHasher(4);
       plainPassword = user.password;
-      user.password = await hashPassword(user.password, 4);
-      jwt_auth_service = new JWTAuthenticationService(userRepo, JWT_SECRET);
+      user.password = await bcrypt_hasher.hashPassword(user.password);
```

Comment:
I am confused about this. Why is it necessary to hash the password here? Shouldn't UserRepository take care of hashing the password? At the moment, it's too easy to forget to hash the password before calling `userRepo.create` and end up with an invalid user record.

Reply:
@bajtos The reason is the "hashing password" logic is handled in the user controller, NOT user repository.
When we create the user by making client request `POST /Users`, the controller will execute the hashing logic to make sure the hashed password is stored.
While when we call `userRepo.create(user)` directly, as what we did in the acceptance test, we need to hash the plain password first then store it.